test/opexecuter.cpp

```diff
@@ -146,15 +146,20 @@ void OpExecuter::CompareNGraphAndTF() {
   for (int i = 0; i < tf_outputs_.size(); i++) {
     switch (expected_output_datatypes_[i]) {
       case DT_FLOAT:
-      case DT_DOUBLE:
         AssertTensorEquals<float>(tf_outputs_[i], ngraph_outputs_[i]);
         break;
       case DT_INT8:
+        AssertTensorEquals<int8>(tf_outputs_[i], ngraph_outputs_[i]);
+        break;
       case DT_INT16:
+        AssertTensorEquals<int16>(tf_outputs_[i], ngraph_outputs_[i]);
+        break;
       case DT_INT32:
-      case DT_INT64:
         AssertTensorEquals<int>(tf_outputs_[i], ngraph_outputs_[i]);
         break;
+      case DT_INT64:
+        AssertTensorEquals<int64>(tf_outputs_[i], ngraph_outputs_[i]);
+        break;
       default:
```

Comment:
The only way to avoid the switch-case, is to know expected_output_datatypes_[i] at compile time
(in which case we would do something like: 
```AssertTensorEquals<EnumToDataType<expected_output_datatypes_[i]>::Type>```

but to know expected_output_datatypes_[i] at runtime, OpExecutor must accept it as template arguments, and not as normal arguments. And since expected_output_datatypes_ is of variable length, it would have to be variadic template inputs to Opexecutor constructor. And variadic templated classes might seriously hurt readability.

Therefore this is currently the best way to do it.

I suggest doing this in each case X:
```EnumToDataType<X>::Type```
to remove the possibility of any error or to future-proof against any type.